appliances/Harbor/appliance.sh

```diff
@@ -0,0 +1,304 @@
+# ---------------------------------------------------------------------------- #
+# Copyright 2024, OpenNebula Project, OpenNebula Systems                  #
+#                                                                              #
+# Licensed under the Apache License, Version 2.0 (the "License"); you may      #
+# not use this file except in compliance with the License. You may obtain      #
+# a copy of the License at                                                     #
+#                                                                              #
+# http://www.apache.org/licenses/LICENSE-2.0                                   #
+#                                                                              #
+# Unless required by applicable law or agreed to in writing, software          #
+# distributed under the License is distributed on an "AS IS" BASIS,            #
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.     #
+# See the License for the specific language governing permissions and          #
+# limitations under the License.                                               #
+# ---------------------------------------------------------------------------- #
+
+# Harbor Docker Registry Appliance for OpenNebula Marketplace
+
+### Utility Functions
+
+gen_db_password() {
+    # In case of password losing, it can be retrieved from the harbor-db container root environment variables 
```

Comment:
What happens if the VM is re-contexted? I'm worried about the password and auto-generated certificates, would anything have a chance to break? :thinking:

Reply:
In case of recontextualization, the db will keep the same password. This password is inside the env variables of a container that is persistently running inside the VM

Reply:
Hmmm I tested it and the certs are not regenerated. What happens exactly after a recontextualization? Should the service_configure() function be triggered again?